feat(changelog): kubernetes-fixed-dns-fix-to-resolve-occasional-resoluti 2024-02-20 (#2774)
scaleway/docs-content: changelog/february2024/2024-02-20-kubernetes-fixed-dns-fix-to-resolve-occasional-resoluti.mdx

---

> title: DNS fix to resolve occasional resolution failures

Reviewer: ```suggestion
title: DNS fix to solve occasional resolution failures
```

status: fixed
author:

>     fullname: 'Join the #[k8s] channel on Slack.'

Reviewer: ```suggestion
    fullname: 'Join the #k8s channel on Slack.'
```

    url: 'https://slack.scaleway.com'
date: 2024-02-20
category: containers
product: kubernetes
---

> The DNS resolution failure has been resolved. To fix the issue, please proceed as follows:
> 
> - Replace or reboot your nodes to get the fix.
> - Use a fqdn to resolve your addresses inside the Private Network. Meaning if you have a `foo` service in the `mypn` pn `foo` will not be resolved and you must use `foo.mypn`
> 
> Important notes:
> - Please avoid using an existing TLD (https://en.wikipedia.org/wiki/Top-level_domain) as PN name if you do so you can have some problem. For instance, if you name your pn `fr` you will not be able to resolve `google.fr` externally, meaning if you have not a service named google in your PN you won’t revsolv `google.fr` and if you have one, it will not be the good one.
> - Please note that `prod` and `dev` are TLD (https://data.iana.org/TLD/tlds-alpha-by-domain.txt)

Reviewer 2: ```suggestion
The DNS resolution issue has been resolved. To implement the fix:

- Replace or reboot your nodes.
- Use a fully qualified domain name (FQDN) for resolving addresses within the Private Network. For example, if you have a service named `foo` in the Private Network `mypn`, use `foo.mypn`.

<Message type="important">
  - Avoid using existing [top-level domains (TLDs)](https://en.wikipedia.org/wiki/Top-level_domain) as Private Network names to prevent potential issues. For instance, naming your PN `fr` may cause problems resolving external addresses like `google.fr`.
  - Be aware that `prod` and `dev` are [valid TLDs](https://data.iana.org/TLD/tlds-alpha-by-domain.txt).
</Message>
```
